mailbox/api/src/test/java/org/apache/james/mailbox/model/UidValidityTest.java

```diff
@@ -38,4 +43,79 @@ void ofShouldReturnValidUidValidity() {
         UidValidity uidValidity = UidValidity.of(expectedValue);
         assertThat(uidValidity.asLong()).isEqualTo(expectedValue);
     }
+
+    @Test
+    void zeroUidValidityShouldBeInvalid() {
+        assertThat(UidValidity.of(0).isValid()).isFalse();
+    }
+
+    @Test
+    void negativeUidValidityShouldBeInvalid() {
+        assertThat(UidValidity.of(-1).isValid()).isFalse();
+    }
+
+    @Test
+    void tooBigUidValidityShouldBeInvalid() {
+        assertThat(UidValidity.of(4294967296L).isValid()).isFalse();
+    }
+
+    @Test
+    void idValidityShouldBeValid() {
+        assertThat(UidValidity.of(42).isValid()).isTrue();
+    }
+
+    @Test
+    void fromSupplierShouldNotThrowWhenZeroIsGenerated() {
+        assertThatCode(() -> UidValidity.fromSupplier(() -> 0L))
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    void fromSupplierShouldNotThrowWhenNegativeIsGenerated() {
+        assertThatCode(() -> UidValidity.fromSupplier(() -> -42L))
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    void fromSupplierShouldNotThrowWhenUpperBoundExclusiveIsGenerated() {
+        assertThatCode(() -> UidValidity.fromSupplier(() -> 4294967296L))
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    void fromSupplierShouldNotThrowWhenUpperBoundInclusiveIsGenerated() {
+        assertThatCode(() -> UidValidity.fromSupplier(() -> 4294967295L))
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    void fromSupplierShouldNotThrowWhenHigherThanUpperBoundIsGenerated() {
+        assertThatCode(() -> UidValidity.fromSupplier(() -> 4294967297L))
+            .doesNotThrowAnyException();
+    }
+
+    @Test
+    void fromSupplierShouldNotThrowWhenLowerThanUpperBoundIsGenerated() {
+        assertThatCode(() -> UidValidity.fromSupplier(() -> 4294967294L))
+            .doesNotThrowAnyException();
+    }
+
+    // Sampling test
+    @RepeatedTest(10000)
+    void randomShouldGenerateValidValues() {
+        assertThatCode(UidValidity::random)
+            .doesNotThrowAnyException();
+    }
+
+    @Disabled("On average upon generating 1.000.000 UidValidity we notice 125 duplicates")
```

Comment:
What are the implications ? Can you state it in the comment ?
And if it is acceptable maybe the test could be one asserting that we have less than 0.0x % of collisions ?